File: Assets/scenes/testing/script/ClampAnimator.cs
using UnityEngine;
using System;


public class ClampAnimator:MonoBehaviour{
	
	public bool test = false;
	public bool triggered = false;

	ParticleSystem[] particles;

	public void Start() {
		particles = GetComponentsInChildren<ParticleSystem>();

	}
	
	public void Update() {
		if(test){
			test = false;
			trigger();
		
		}
	}
	
	public void trigger(){
		if(triggered){ return; };
		triggered = true;
		foreach (ParticleSystem p in particles){
			p.Stop();
			p.Play();
		}
		foreach(Animation a in GetComponentsInChildren<Animation>()){
			a.Play("open");
		}
	}
}

File: Assets/scenes/testing/script/ClampAnimator.js
#pragma strict

var test : boolean = false;
var triggered : boolean = false;
function Start () {

}

function Update () {
	if(test){
		test = false;
		trigger();
	
	}
}

function trigger(){
	if(triggered){ return; };
	triggered = true;
	for(var a : Animation in GetComponentsInChildren.<Animation>()){
		a.Play("open");
	}
}